### PR TITLE
turbolinks accepts a document, run tests inside an iframe

### LINKED ISSUE #138: Turbolinks should accept a document parameter
Turbolinks should allow you to set a document for it to operate on. This would allow it to control iframes and the like, or possibly do it's manipulations and then have something like morphdom perform a diff afterwards, as well as allow tests to be done using a document fragment, thus making Teaspoon's test runner more useful.

### PR DESCRIPTION
**Pull request by TheMallen**
## We now have a browser runner again!
image
(closes #138)

This PR makes two primary changes.
- `Turbolinks.document` allows you to set a document for turbolinks to operate on.
- Tests are now run leveraging this, allowing for use of the browser test runner by sandboxing TG's DOM mutations to an iframe.

## How
We had to move the way we defined ASSET_FIXTURES due to some weirdness in teaspoon config, and change all uses of `document` to reference a variable in module scope to facilitate this change. We also had to make some minor changes in how we initialize tests.

## Why
Development velocity is better when we can actually use the browser test runner. Everything else is gravy.

## Follow up:
We should change `remote_test` to do this as well, since it's currently suffering from the same problem `turbolinks_test` had.

## Reviewers
@qq99 
@GoodForOneFare 
@edward 
CC @Shopify/tnt

### CODE REVIEW

**Comment by qq99** [comment]
Nice :D

File: test/javascripts/turbolinks_test.coffee [hunk 1]
```diff
@@ -91,23 +101,32 @@ describe 'Turbolinks', ->
...
     $("#turbo-area").remove()
 
   it 'is defined', ->
     assert(Turbolinks)
 
+  describe '#document', ->
+    afterEach ->
+      Turbolinks.document(testDocument)
+
+    it 'assigning a document', ->
```
Comment (GoodForOneFare) [inline test/javascripts/turbolinks_test.coffee]: Rename to state some kind of behaviour, or cause and effect.  Or just delete.

**Comment by GoodForOneFare** [comment]
I can run browser tests!  😃 Excellent stuff 👍

File: test/javascripts/turbolinks_test.coffee (hunk 1 above)
Comment (TheMallen) [inline test/javascripts/turbolinks_test.coffee]: I'll delete it :P

**Comment by TheMallen** [comment]
@qq99 You think this is good to 🚢 ?

File: lib/assets/javascripts/turbograft/turbolinks.coffee [hunk 2]
```diff
@@ -42,7 +43,7 @@ requestMethodIsSafe =
 browserSupportsTurbolinks = browserSupportsPushState and requestMethodIsSafe
 
 browserSupportsCustomEvents =
-  document.addEventListener and document.createEvent
+  activeDocument.addEventListener and activeDocument.createEvent
```
Comment (qq99) [inline lib/assets/javascripts/turbograft/turbolinks.coffee]: this could probably just be `document` right?

**Pull request merged**